benchmark/single-source/SetTests.swift

```diff
@@ -12,107 +12,156 @@
+    tags: [.validation, .api, .Set],
+    setUpFunction: { blackHole([setOAB, setOBC]) }),
+
+  BenchmarkInfo(
+    name: "SetIntersect",
+    runFunction: { n in run_SetIntersect(setAB, setCD, 0, 100 * n) },
+    tags: [.validation, .api, .Set],
+    setUpFunction: { blackHole([setAB, setCD]) }),
+  BenchmarkInfo(
+    name: "SetIntersect2",
+    runFunction: { n in run_SetIntersect(setAB, setBC, countB, 10 * n) },
+    tags: [.validation, .api, .Set],
+    setUpFunction: { blackHole([setAB, setBC]) }),
+  BenchmarkInfo(
+    name: "SetIntersect_OfObjects",
+    runFunction: { n in run_SetIntersect_OfObjects(setOAB, setOCD, 0, 100 * n) },
+    tags: [.validation, .api, .Set],
+    setUpFunction: { blackHole([setOAB, setOCD]) }),
+  BenchmarkInfo(
+    name: "SetIntersect2_OfObjects",
+    runFunction: { n in run_SetIntersect_OfObjects(setOAB, setOBC, countB, 10 * n) },
+    tags: [.validation, .api, .Set],
+    setUpFunction: { blackHole([setOAB, setOBC]) }),
+
+  BenchmarkInfo(
+    name: "SetIsSubsetOf",
+    runFunction: { n in run_SetIsSubsetOf(setAB, setCD, false, 5000 * n) },
+    tags: [.validation, .api, .Set],
+    setUpFunction: { blackHole([setAB, setCD]) }),
+  BenchmarkInfo(
+    name: "SetIsSubsetOf2",
+    runFunction: { n in run_SetIsSubsetOf(setB, setAB, true, 50 * n) },
+    tags: [.validation, .api, .Set],
+    setUpFunction: { blackHole([setB, setAB]) }),
+  BenchmarkInfo(
+    name: "SetIsSubsetOf_OfObjects",
+    runFunction: { n in run_SetIsSubsetOf_OfObjects(setOAB, setOCD, false, 5000 * n) },
+    tags: [.validation, .api, .Set],
+    setUpFunction: { blackHole([setOAB, setOCD]) }),
+  BenchmarkInfo(
+    name: "SetIsSubsetOf2_OfObjects",
+    runFunction: { n in run_SetIsSubsetOf_OfObjects(setOB, setOAB, true, 50 * n) },
+    tags: [.validation, .api, .Set],
+    setUpFunction: { blackHole([setOB, setOAB]) }),
+
+  BenchmarkInfo(
+    name: "SetUnion",
+    runFunction: { n in run_SetUnion(setAB, setCD, countABCD, 100 * n) },
+    tags: [.validation, .api, .Set],
+    setUpFunction: { blackHole([setAB, setCD]) }),
+  BenchmarkInfo(
+    name: "SetUnion2",
+    runFunction: { n in run_SetUnion(setAB, setBC, countABC, 10 * n) },
+    tags: [.validation, .api, .Set],
+    setUpFunction: { blackHole([setAB, setBC]) }),
+  BenchmarkInfo(
+    name: "SetUnion_OfObjects",
+    runFunction: { n in run_SetUnion_OfObjects(setOAB, setOCD, countABCD, 100 * n) },
+    tags: [.validation, .api, .Set],
+    setUpFunction: { blackHole([setOAB, setOCD]) }),
+  BenchmarkInfo(
+    name: "SetUnion2_OfObjects",
+    runFunction: { n in run_SetUnion_OfObjects(setOAB, setOBC, countABC, 10 * n) },
+    tags: [.validation, .api, .Set],
+    setUpFunction: { blackHole([setOAB, setOBC]) }),
+]
 
 @inline(never)
-func sink(_ s: inout Set<Int>) {
+public func run_SetIsSubsetOf(
+  _ a: Set<Int>,
+  _ b: Set<Int>,
+  _ r: Bool,
+  _ n: Int) {
+  for _ in 0 ..< n {
+    let isSubset = a.isSubset(of: identity(b))
```

Comment:
One more thing… could you document what's the purpose of wrapping `b` in `identity` here?

Reply:
It should be a relatively common idiom in these benchmarks -- it's there to prevent the compiler from moving things out of the loop. `a` and `b` are constant through all iterations, and in theory, a sufficiently smart compiler could optimize some/all of it away. Adding an opaque function call with unknown effects (hopefully) defeats these optimizations.

(I don't think any optimizations would apply in these cases, but it's better to be safe than sorry.)

Reply:
That explanation makes perfect sense. How about putting it in the source comment? That way it’s easier to point people unfamiliar with the idiom to this file as an example of best practice (link from docs, when we get to it).